### PR TITLE
topology2: ssp: add quirks for NHLT table endpoints

### PR DESCRIPTION
**Pull request by brentlu**
Two quirk flags are added. The quirk "bt_sideband" will set the device type of corresponding NHLT endpoints to BT(4) and others to Codec(0).

The other quirk "render_feedback" is for speaker path with echo reference or IV sense feedback. The endpoint direction will use Render with loopback(3) and Feedback for render(4) instead of Render(0) and Capture(1).

### CODE REVIEW

**Comment by brentlu** [comment]
PR:
linux: https://github.com/thesofproject/linux/pull/4492
alsa-utils: https://github.com/alsa-project/alsa-utils/pull/226
topology: https://github.com/thesofproject/sof/pull/7984

File: tools/topology/topology2/include/dais/ssp.conf [hunk 1]
```diff
@@ -91,9 +91,17 @@ Class.Dai."SSP" {
 		constraints {
 			!valid_values [
 				"lbm_mode"
+				"bt_sideband"
+				"render_feedback"
+				"lbm_mode,bt_sideband"
+				"lbm_mode,render_feedback"
```
Comment (plbossart) [inline tools/topology/topology2/include/dais/ssp.conf:97]: this doesn't scale, we don't want to create a new definition for each possible combination. This needs to be a mask or something.

File: tools/topology/topology2/include/dais/ssp.conf [hunk 2]
```diff
@@ -91,9 +91,17 @@ Class.Dai."SSP" {
...
 			]
 			!tuple_values [
 				64
+				128
+				256
+				192
+				320
```
Comment (plbossart) [inline tools/topology/topology2/include/dais/ssp.conf:104]: this doesn't seem related to NHLT?

**Review by plbossart — CHANGES_REQUESTED** [review]

File: tools/topology/topology2/include/dais/ssp.conf (hunk 1 above)
Comment (brentlu) [reply tools/topology/topology2/include/dais/ssp.conf:97]: I know this change looks stupid... Another way is to add new property like 'nhlt_quirk' to handle this.

Comment (plbossart) [reply tools/topology/topology2/include/dais/ssp.conf:97]: even if we add a new property, what we have today is 'global' and has no notion of direction, but the NHLT tags are not-only endpoint specific but direction specific. I am not sure how to reconcile the two, we have no placeholder for direction-specific information.

Comment (lgirdwood) [reply tools/topology/topology2/include/dais/ssp.conf:97]: I think here we could have the NHLT object and only assign direction to it when we instanciate it ? @ranj063 ?
